diff -ru scinterm_1.6-org/Makefile scinterm_1.6/Makefile
--- scinterm_1.6-org/Makefile	2015-04-19 03:22:28.000000000 +0900
+++ scinterm_1.6/Makefile	2016-01-17 19:34:31.000000000 +0900
@@ -5,7 +5,7 @@
 AR = ar
 CXX = g++
 INCLUDEDIRS = -I ../include -I ../src -I ../lexlib
-CXXFLAGS = -Wall -pedantic -DCURSES -DSCI_LEXER $(INCLUDEDIRS)
+CXXFLAGS = -Wall -pedantic -DCURSES -DSCI_LEXER $(INCLUDEDIRS) -D_XOPEN_SOURCE_EXTENDED
 ifdef DEBUG
   CXXFLAGS += -DDEBUG -g
 else
diff -ru scinterm_1.6-org/ScintillaTerm.cxx scinterm_1.6/ScintillaTerm.cxx
--- scinterm_1.6-org/ScintillaTerm.cxx	2015-04-19 03:22:28.000000000 +0900
+++ scinterm_1.6/ScintillaTerm.cxx	2016-01-17 19:34:46.000000000 +0900
@@ -7,6 +7,7 @@
 
 #include <math.h>
 #include <stdint.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <stdexcept>
@@ -417,7 +418,13 @@
   void MeasureWidths(Font &font_, const char *s, int len,
                      XYPOSITION *positions) {
     for (int i = 0, j = 0; i < len; i++) {
-      if (!UTF8IsTrailByte((unsigned char)s[i])) j++;
+      if (!UTF8IsTrailByte((unsigned char)s[i])) {
+	if (UTF8IsAscii((unsigned char)s[i])) {
+          j++;
+        } else {
+          j += 2;
+        }
+      }
       positions[i] = j;
     }
   }
